### ISSUE TITLE
[Routing] Cache limit

### ISSUE THREAD

**Issue by aaronadal** [symfony/symfony#29614]
**Symfony version(s) affected**: 4.2.1

**Description**  
I'm having trouble while upgrading a big project from 3.4 to flex and I think the problem resides in the routing cache files. When the *srcApp_KernelDevDebugContainerUrlMatcher.php* cached file is loaded, an ERR_CONNECTION_RESET is shown in Google Chrome. I tried to import it (removing all the dependencies) from a raw PHP file and the error persists. I think it is too huge and PHP is unable to include it. If I manually remove some lines from this file, it works well.

**How to reproduce**  
I've created a dummy gist here: https://gist.github.com/aaronadal/293b6951c212fc1d575a132f2a667aab
I can't try it in another computer, so if the limit depends on the computer itself, it could be necessary to add extra lines of code to the included class constructor to reproduce the error.

**Possible Solution**  
Rethinking of caching method or possibility to go back to the previous, slow version.

**Additional context**  
Windows 10, PHP 7.2.1, Apache 2.4.29

**Comment by chiqui3d** [comment]
Something has to have bad in your routes, check that you have everything well following the annotations and guidelines of 4.2, because I see that your file Router Cache file generates this many times:

```
.'|/_(?'
                        .'|error/(\\d+)(?:\\.([^/]++))?(*:38)'
                        .'|wdt/([^/]++)(*:57)'
                        .'|profiler/([^/]++)(?'
                            .'|/(?'
                                .'|search/results(*:102)'
                                .'|router(*:116)'
                                .'|exception(?'
                                    .'|(*:136)'
                                    .'|\\.css(*:149)'
                                .')'
                            .')'
                            .'|(*:159)'
                        .')'
                    .')'
```

**Comment by nicolas-grekas** [comment]
Can you share a small app that generates such cache for the router matcher?

**Comment by aaronadal** [comment]
Sorry if I was not clear before. The code in the gist is not real. That code was not generated by Symfony, I just simulated it . But I think it doesn't matters. The result is the same.

The more routes an application has, the more lines of code the cached file has. And when the file grows too much, the require_once statement fails. I don't know why, but it fails. 

The gist consists of two files: one of them requires the other. The error occurs when requiring the second file. However, if you try to make that second file smaller, the require statement is executed successfully. I think it is a PHP limitation, but I couldn't find information about it. Doesn't it happen to you?
    
I hope it is clearer now.

**Comment by chiqui3d** [comment]
Can it be because of the php.ini settings? 
https://stackoverflow.com/questions/40458863/php-err-connection-reset-error-with-file-upload-more-than-8mbs

**Comment by alexislefebvre** [comment]
The gist doesn't mention Symfony, so the error is only related to PHP? You may ask this question on Stack Overflow (with `PHP` tag, not `symfony`).

**Comment by nicolas-grekas** [comment]
We would need a reproducer in order to find a workaround. Can you try increasing the opcache related ini settings? If you can share the real compiled matcher (privately) I may have a look also.

**Comment by aaronadal** [comment]
This weekend I ran the code in other computers and it worked fine, so the error seems to be in my configuration. I've asked in StackOverflow as you recommend me: [ERR_CONNECTION_RESET when require_once a file](https://stackoverflow.com/questions/53812357/err-connection-reset-when-require-once-a-file).

I thought it could be a generalized PHP error, so I opened this issue. I think it can be closed now. Sorry for the inconvenience.

And thank you very much.
